remix-tests/src/testRunner.ts

```diff
@@ -19,25 +19,53 @@ function getOverridedSender (userdoc, signature: string, methodIdentifiers) {
     return fullName && accountIndex ? accountIndex[1] : null
 }
 
-function getAvailableFunctions (jsonInterface) {
-    return jsonInterface.reverse().filter((x) => x.type === 'function').map((x) => x.name)
+/**
+ * @dev returns functions of a test contract file in same sequence they appear in file (using passed AST)
+ * @param fileAST AST of test contract file source
+ * @param testContractName Name of test contract
+ */
+
+function getAvailableFunctions (fileAST: any, testContractName: string) {
+    const contractAST: any[] = fileAST.nodes.filter(node => node.name === testContractName && node.nodeType === 'ContractDefinition')
+    const funcNodes: any[] = contractAST[0].nodes.filter(node => node.kind === 'function' && node.nodeType === "FunctionDefinition")
+    const funcList: string[] = funcNodes.map(node => node.name)
+    return funcList;
 }
 
-function getTestFunctions (jsonInterface) {
-    let specialFunctions = ['beforeAll', 'beforeEach', 'afterAll', 'afterEach']
-    return jsonInterface.filter((x) => specialFunctions.indexOf(x.name) < 0 && x.type === 'function')
+/**
+ * @dev returns ABI of passed method list from passed interface
+ * @param jsonInterface Json Interface
+ * @param funcList Methods to extract the interface of
+ */
+
+function getTestFunctionsInterface (jsonInterface: any, funcList: string[]) {
+    let functionsInterface: any[] = []
+    const specialFunctions = ['beforeAll', 'beforeEach', 'afterAll', 'afterEach']
+    for(const func of funcList){
+        if(!specialFunctions.includes(func))
+            functionsInterface.push(jsonInterface.find(node => node.type === 'function' && node.name === func))
```

Comment:
Create a variable for `jsonInterface.find(node => node.type === 'function' && node.name === func)` with a name that describe what it is.